backend/sublet/serializers.py

```diff
@@ -0,0 +1,93 @@
+from rest_framework import serializers
+
+from sublet.models import Amenity, Offer, Sublet
+
+
+class AmenitySerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Amenity
+        fields = "__all__"
+
+
+class OfferSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Offer
+        fields = "__all__"
+        read_only_fields = ["id", "created_date", "user"]
+
+    def create(self, validated_data):
+        validated_data["user"] = self.context["request"].user
+        return super().create(validated_data)
+
+
+# complex sublet serializer for use in C/U/D + getting info about a singular sublet
+class SubletSerializer(serializers.ModelSerializer):
+    amenities = AmenitySerializer(many=True, required=False)
+    # sublettees = OfferSerializer(many=True, required=False, read_only=True)
+
+    class Meta:
+        model = Sublet
+        exclude = ["favorites"]
+        read_only_fields = ["id", "created_at", "subletter", "sublettees"]
+
+    def parse_amenities(self, raw_amenities):
+        if isinstance(raw_amenities, list):
+            ids = raw_amenities
+        else:
+            ids = (
+                list() if len(raw_amenities) == 0 else [str(id) for id in raw_amenities.split(",")]
```

Comment:
why do `str(id)`? 
I guess my confusion is if you're doing .split, raw_amenities is already a string? did you mean to do int/did this work when u tested it?

Reply:
It works. To be honest I'm not exactly sure how this works but this is a django `QueryDict` which is part of `request.POST` (not `request.data`). Very bad syntax, we should probably change, but this is an alternative way to add amenities to the sublet